packages/backend/src/services/PromoteService/PromoteService.ts

```diff
@@ -1053,46 +1121,13 @@ export class PromoteService extends BaseService {
         });
 
         const chartChanges: PromotionChanges['charts'] = charts.map(
-            ({ promotedChart, upstreamChart }) => {
-                if (upstreamChart.chart !== undefined) {
-                    // TODO check differences to see if we need to UPDATE or NO_CHANGES
-                    // For this we'll have to fetch more data for upstreamChart like  the chart config
-                    // or check if promotedChart.updatedAt > upstreamChart.updatedAt
-                    const isEqual = false; // deepEqual(promotedChart.chart, upstreamChart.chart)
-
-                    return {
-                        action: isEqual
-                            ? PromotionAction.NO_CHANGES
-                            : PromotionAction.UPDATE,
-                        data: {
-                            ...promotedChart.chart,
-                            dashboardUuid: promotedChart.chart.dashboardUuid, // change the dashboard uuid after creation
-                            spaceUuid: upstreamChart.chart.spaceUuid,
-                            uuid: upstreamChart.chart.uuid,
-                            spaceSlug: promotedChart.space?.slug,
-                            oldUuid: promotedChart.chart.uuid,
-                        },
-                    };
-                }
-                return {
-                    action: PromotionAction.CREATE,
-                    data: {
-                        ...promotedChart.chart,
-                        dashboardUuid:
-                            upstreamDashboard.dashboard?.uuid ||
-                            promotedChart.chart.dashboardUuid, // set the new space uuid after creation
-                        spaceUuid:
-                            upstreamChart.space?.uuid ||
-                            promotedChart.space.uuid, // set the new space uuid after creation
-                        projectUuid: upstreamProjectUuid,
-                        spaceSlug: promotedChart.space?.slug,
-                        oldUuid: promotedChart.chart.uuid,
-                    },
-                };
-            },
+            ({ promotedChart, upstreamChart }) =>
+                PromoteService.getChartChange(promotedChart, upstreamChart),
         );
 
         // TODO return charts within dashboards that are going to be deleted after the promotion
+        // For this we'll need to get all the tiles for the upstreamDashboard and compare against the promotedDashboard.tiles
```

Comment:
I tried to implement this, but we are also missing upstreamDashboard.tiles, so not for now